src/Microsoft.Health.Fhir.CosmosDb/Registration/FhirServerBuilderCosmosDbRegistrationExtensions.cs

```diff
@@ -181,7 +181,21 @@ private static IFhirServerBuilder AddCosmosDbPersistence(this IFhirServerBuilder
                 .Transient()
                 .AsService<ICollectionDataUpdater>();
 
+            services.Add<ResourceManagerCollectionSetup>()
+                .Singleton()
+                .AsSelf();
+
             services.Add<DataPlaneCollectionSetup>()
+                .Singleton()
+                .AsSelf();
+
+            services.Add(c =>
+                {
+                    CosmosDataStoreConfiguration config = c.GetRequiredService<CosmosDataStoreConfiguration>();
+                    return config.UseManagedIdentity
+                        ? (ICollectionSetup)c.GetRequiredService<ResourceManagerCollectionSetup>()
```

Comment:
Would ResourceManagerCollectionSetup always be invoked? 
I think for OSS it should, thinking more in terms of SF and AKS?